libs/wire-api/src/Wire/API/User/Client.hs

```diff
@@ -97,6 +97,92 @@ import Wire.API.Arbitrary (Arbitrary (arbitrary), GenericUniform (..), generateE
+-- type), and the two approaches could be used in parallel indefinitely.
+--
+-- Feature flags only reveal the minimum amount of information necessary to handle the client,
+-- making it harder to fingerprint and track clients; they are straight-forward and
+-- self-documenting (to an extent), and make it easier to release a feature on the backend and
+-- clients independently.
+--
+-- Platform/version info is if you have many different feature switches, even though it
+-- doesn't solve the problem of having to explore the entire feature space in your tests.
+-- They give you a better idea of the time line, and how to gently discontinue support for
+-- ancient features.
+data SupportedClientFeature
+  = -- | Clients have minimum support for LH, but not for explicit consent.  Implicit consent
+    -- is granted via the galley server config (see '_setLegalHoldTeamsWhitelist').
+    ClientSupportsLegalholdImplicitConsent
+  deriving stock (Eq, Ord, Bounded, Enum, Show, Generic)
+  deriving (Arbitrary) via (GenericUniform SupportedClientFeature)
+  deriving (ToJSON, FromJSON, Swagger.ToSchema) via S.Schema SupportedClientFeature
+
+instance S.ToSchema SupportedClientFeature where
+  schema =
+    S.named "SupportedClientFeature" $
+      S.tag _ClientSupportsLegalholdImplicitConsent S.schema
+
+makePrisms ''SupportedClientFeature
+
+typeSupportedClientFeature :: Doc.DataType
+typeSupportedClientFeature =
+  Doc.string $
+    Doc.enum
+      [ "legalhold-implicit-consent"
+      ]
+
+instance Cql.Cql SupportedClientFeature where
+  ctype = Cql.Tagged Cql.IntColumn
+
+  toCql ClientSupportsLegalholdImplicitConsent = Cql.CqlInt 1
+
+  fromCql (Cql.CqlInt i) = case i of
+    1 -> return ClientSupportsLegalholdImplicitConsent
+    n -> Left $ "Unexpected SupportedClientFeature value: " ++ show n
+  fromCql _ = Left "SupportedClientFeature value: int expected"
+
+-- FUTUREWORK: add golden tests for this?
+data SupportedClientFeatureList = SupportedClientFeatureList {fromSupportedClientFeatureList :: [SupportedClientFeature]}
+  deriving stock (Eq, Ord, Show, Generic)
+  deriving (Arbitrary) via (GenericUniform SupportedClientFeatureList)
+  deriving (ToSchema) via (CustomSwagger '[FieldLabelModifier (StripPrefix "fromSupportedClient", CamelToSnake)] SupportedClientFeatureList)
+
+modelClientSupportedFeatureList :: Doc.Model
+modelClientSupportedFeatureList = Doc.defineModel "SupportedClientFeatureList" $ do
+  Doc.description "Hints provided by the client for the backend so it can behavior in a backwards-compatible way."
+  Doc.property "feature_list" (Doc.array typeSupportedClientFeature) $ do
+    Doc.description "Array containing all supported features."
+
+instance ToJSON SupportedClientFeatureList where
+  toJSON (SupportedClientFeatureList l) = object ["feature_list" .= l]
+
+instance FromJSON SupportedClientFeatureList where
+  parseJSON = withObject "SupportedClientFeatureList" $ \obj -> SupportedClientFeatureList <$> obj .: "feature_list"
+
```

Comment:
These can be derived automatically if you write an `S.ToSchema` instance. Something like the following should work (haven't tried compiling it though):
```haskell
S.object "SupportedClientFeatureList" $
  fromSupportedClientFeatureList S..= S.field "feature_list" S.schema
```